### ISSUE TITLE
recipe-writing for UC30.recipe1 (FASTQ specification)

### ISSUE THREAD

**Issue by robertgiessmann** [FAIRplus/the-fair-cookbook#74]
Hi Philippe and extended editorial board,

here comes the abstract for UC30.recipe1 on the publicly available FASTQ specifications:

[**Ingredient description: understanding FASTQ format and its specification**](https://github.com/FAIRplus/the-fair-cookbook/blob/uc30/docs/content/ingredients/ingredient-description-fastq-spec.md)

The FASTQ format is a popular format for storing sequences (i.e. letters representing nucleotides in a piece of DNA) and their corresponding quality scores. FASTQ is a text-based data format with weak specification. This ingredient description will exemplify how a FASTQ file can look like, what variants=dialects are commonly found, and will list some sources for weak=pseudo specifications of this important de facto standard file format.

Here is the link to the recipe: https://github.com/FAIRplus/the-fair-cookbook/blob/uc30/docs/content/ingredients/ingredient-description-fastq-spec.md

Dear Philippe and others, do you have feedback on this abstract?

Thanks,
   Robert

**Comment by robertgiessmann** [comment]
https://en.wikipedia.org/wiki/FASTQ_format

**Comment by robertgiessmann** [comment]
The Sanger FASTQ file format for sequences with quality scores, and the
Solexa/Illumina FASTQ variants.
P.J.A.Cock (Biopython), C.J.Fields (BioPerl), N.Goto (BioRuby),
M.L.Heuer (BioJava) and P.M. Rice (EMBOSS).
Nucleic Acids Research 2010 38(6):1767-1771
https://doi.org/10.1093/nar/gkp1137

**Comment by robertgiessmann** [comment]
time tracking (tt): maybe 3 hours of total thought time spend on the whole use case (before); until now:
  - 15 min for writing of abstract (plus time for thoughts while going running: ~30min)
  - 20 min for setting up the skeleton structure for both recipes

**Comment by robertgiessmann** [comment]
tt: 
- 40 min of writing and reading NAR paper

**Comment by robertgiessmann** [comment]
tt:
  - 10 min of additional styling of the ingredient description

**Comment by robertgiessmann** [comment]
please find the recipe draft here: https://github.com/FAIRplus/the-fair-cookbook/blob/uc30/docs/content/ingredients/ingredient-description-fastq-spec.md

**Comment by proccaserra** [comment]
I would the FAIRsharing record: https://fairsharing.org/FAIRsharing.r2ts5t

It would interested to obtain form @tburdett the list of tools used by ENA/EGA/AE for dealing/checking FASTQ files. As main consumers/providers, it will give an indication of trust

**Comment by robertgiessmann** [comment]
I agree, fairsharing should be included. While item 2 is principally interesting, I wouldn't need it for this use case, plus it introduces just another dependency. If desired by tburdett or someone else from EMBL-EBI, the recipe is of course open for future contributions.

**Comment by FuqiX** [comment]
> I would the FAIRsharing record: https://fairsharing.org/FAIRsharing.r2ts5t
> 
> It would interested to obtain form @tburdett the list of tools used by ENA/EGA/AE for dealing/checking FASTQ files. As main consumers/providers, it will give an indication of trust

ENA used to use [FASTQ_utils](https://github.com/nunofonseca/fastq_utils) for FASTQ validation. Fastq_utils is also used by many external groups.  ENA is also developing a new FASTQ validator. I am checking with them.

There’ are some compromises to do this: paired reads should ideally be validated together, for example.

ENA also have their FASTQ file specifications [here](https://ena-docs.readthedocs.io/en/latest/submit/fileprep/reads.html#fastq-format)

**Comment by robertgiessmann** [comment]
Hi Fuqi, great new use cases about ENA -- so far, that (incl. paired reads) was not applicable to my initial idea, but great to keep it in mind.

**Comment by robertgiessmann** [comment]
Here is the link to the recipe: https://github.com/FAIRplus/the-fair-cookbook/blob/uc30/docs/content/ingredients/ingredient-description-fastq-spec.md

**Comment by tburdett** [comment]
> I would interested to obtain form @tburdett the list of tools used by ENA/EGA/AE for dealing/checking FASTQ files. As main consumers/providers, it will give an indication of trust

FASTQ_utils, linked to by @FuqiX, is indeed commonly used within EBI. Currently ENA (and EGA) uses the [Samtools htsjdk](https://github.com/samtools/htsjdk) library for FASTQ validation, but with additional custom logic for checking the correct sets of files are received depending on the technology. Fuqi already described the paired reads scenario, but there are others: for example, for 10x sequencing we require BAM/CRAM rather than FASTQ because of the potential for missing barcodes. This is a longer conversation, though, which may be out of scope for this recipe. 

**TL;DR:** FASTQ validation is often inadequate without also at least a minimal definition of the technology used to generate the sequence being encoded.
